### PR TITLE
[jbehave-core] add ignore/comment logic into addComposedStep() method

### PR DESCRIPTION
**Pull request by VitaliLebedzeu**
Change if block's variables in addComposedSteps() method and ignore/comment logic to addComposedStep() method
https://github.com/vividus-framework/vividus/issues/649

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java [hunk 1]
```diff
@@ -198,25 +198,35 @@ public void addComposedSteps(List<Step> steps, String stepAsString, Map<String,
...
-                // candidate is itself composite: recursively add composed steps
-                candidate.addComposedSteps(composedSteps, composedStep, matchedParameters, allCandidates);
-            }
-            steps.add(candidate.createMatchedStep(composedStep, matchedParameters, composedSteps));
+        if (ignore(composedStep)) {
+            // ignorable steps are added so they can be reported
+            steps.add(StepCreator.createIgnorableStep(composedStep));
         } else {
-            steps.add(StepCreator.createPendingStep(composedStep, previousNonAndStep));
+            if (comment(composedStep)) {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java]: `} else if (comment(composedStep)) {`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java (hunk 1 above)
Comment (VitaliLebedzeu) [reply jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java]: My fail... will be fixed.

Comment (VitaliLebedzeu) [reply jbehave-core/src/main/java/org/jbehave/core/steps/StepCandidate.java]: fixed

File: jbehave-core/src/test/java/org/jbehave/core/steps/CompositeCandidateStepsBehaviour.java [hunk 2]
```diff
@@ -312,8 +313,58 @@ public void typePassword(){}
...
+        Mockito.verify(reporter).ignorable("!-- When ignore me");
+    }
+
+    static class CompositeWithIgnorableStep extends Steps {
+
+        @When("Ignore my step")
+        @Composite(steps={"!-- When ignore me"})
+        public void whenIgnoreMyStep(){}
+
+        @When("When ignore me")
```
Comment (uarlouski) [inline jbehave-core/src/test/java/org/jbehave/core/steps/CompositeCandidateStepsBehaviour.java]: just @When("ignore me")

**Review by uarlouski — APPROVED** [review]

File: jbehave-core/src/test/java/org/jbehave/core/steps/CompositeCandidateStepsBehaviour.java (hunk 2 above)
Comment (VitaliLebedzeu) [reply jbehave-core/src/test/java/org/jbehave/core/steps/CompositeCandidateStepsBehaviour.java]: fixed

**Comment by valfirst** [comment]
This fix was merged to `master` branch of upstream repository

**Pull request closed without merging**